tests/models/test_responses.py

```diff
@@ -639,7 +639,7 @@ def test_iter_lines():
         content=b"Hello,\nworld!",
     )
     content = [line for line in response.iter_lines()]
-    assert content == ["Hello,\n", "world!"]
+    assert content == ["Hello,", "world!"]
```

Comment:
These test changes show that currently this PR introduces a behavior change. I assume we don't want behavior to change, i.e. for `\n` to stay in the yielded lines?

Is this as straight-forward as doing `splitlines(keep_ends=True)`... ?

Reply:
Oh, I didn't realise there was a `keepends` flag, let me rework the patch with that.

Reply:
Right, the issue is that `keepends` converts them all to `\n`